### PR TITLE
Update to add documentation generation and revisions regarding Upjet

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

This should match the new documentation revision proposed in https://github.com/upbound/upjet/pull/115

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Went through with https://github.com/upbound/upjet/pull/115

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/v1beta1/types.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ type ProviderConfigStatus struct {
 
 // +kubebuilder:object:root=true
 
-// A ProviderConfig configures a Template JET provider.
+// A ProviderConfig configures a Template provider.
```
Comment (turkenh) [inline apis/v1beta1/types.go:35]: Should we also change the category below from `templatejet` to `template` ?

**Review by turkenh — APPROVED** [review]
Thanks @muvaf !

File: apis/v1beta1/types.go (hunk 1 above)
Comment (muvaf) [reply apis/v1beta1/types.go:35]: Thanks for catching!

File: hack/prepare.sh [hunk 2]
```diff
@@ -1,25 +1,29 @@
 #!/usr/bin/env bash
+set -euoX pipefail
```
Comment (turkenh) [inline hack/prepare.sh:2]: Getting the following on MAC (zsh):
```
./hack/prepare.sh: line 2: set: -X: invalid option
set: usage: set [--abefhkmnptuvxBCHP] [-o option] [arg ...]
```

**Pull request merged**